### DISCUSSION TITLE
How does Joomla4 detect Compatibility of extension

### DISCUSSION THREAD

**Discussion by jschmi102** [joomla/joomla-cms#34664] [Q&A]
Just did install Joomla 4 RC3 and was wondering how Joomla updater does checks if an extension is compatibile with current Joomla version. Is it the extenson Info used which is stored in Joomla extensions directory? Is the extension's xml checked for specifics, Or will the the xml-version setting influence the behaviour?

**Comment by richard67** [comment]
@zero-24 Could you step in and explain how the compatibility check works, or link to some docs which do? Thanks in advance.

**Comment by zero-24** [comment]
Hi, its the update xml thats checked whether the extension is compatible or not: https://docs.joomla.org/Deploying_an_Update_Server
https://docs.joomla.org/Pre-Update_Check

**Reply by brianteeman** [reply]
Hi, its the update xml thats checked whether the extension **declares that it is** compatible or not

**Comment by Bakual** [comment]
As I just reverse-engineered it for my extension: The checker individually checks for each extensions its update manifest (the XML on the developers site) to see if there is a version listed for Joomla 4.0. It is not using the cached update table. There is also JavaScript involved to do that.
It seems to have a few differences (issues) to the regular update check. For example it requires the `client` tag for components, which the regular updater doesn't require (defaults to administrator). And you need to specifiy the targetversion as `4.[0]`, `4.0` doesn't work.

**Comment by jschmi102** [comment]
hi,
Thanks for your response - now I do understand.. (had to change 4.0 to 4.[0].
